app/assets/javascripts/hidden_values.js

```diff
@@ -5,12 +5,12 @@ function turn_textarea_switch(checkbox) {
   var $editorContainer = $('.editor-container');
 
   if (checkbox.checked) {
-    target = '<input class="form-control" type="password" id="' + id + '" name="' + source.name + '" value ="' + source.value + '"></input>'
+    target = '<input class="form-control" type="password" id="' + id + '" name="' + source.name + '" value ="' + _.escape($(source).val()) + '"></input>'
```

Comment:
Sorry, my comment probably wasn't clear.  I meant that we usually use .val() to *set* the value of an element to avoid HTML escaping, so we'd create the input field, then call input.val(html_text).